src/Features/Core/Portable/LanguageServices/SymbolDisplayService/AbstractSymbolDisplayService.AbstractSymbolDescriptionBuilder.cs

```diff
@@ -198,7 +198,11 @@ private void AddExceptions(ISymbol symbol)
             {
                 if (symbol.GetAttributes().Any(x => x.AttributeClass.MetadataName == "ObsoleteAttribute"))
                 {
-                    AddDeprecatedPrefix();
+                    var typeSymbol = symbol as ITypeSymbol;
+                    if (typeSymbol == null || !typeSymbol.IsByRefLikeType)
```

Comment:
One issue with this approach is that a `ref struct` with a user-provided Obsolete attribute won't show either. I don't feel strongly on that, but wanted to call it out for IDE team.

Reply:
Is the compiler expected to elide the attribute out of the GetAttributes() call? Do we have prior art here regarding INamedTypeSymbol.IsComImport or anything else?

Reply:
I also think omitting special obsolete in getattributes might be cleaner.